### PR TITLE
Add option to use go internal modules in github actions

### PR DESCRIPTION
**Pull request by KacperMalachowski**
**Description**

Changes proposed in this pull request:

- Add option to run image builder with access to go internal modules

### CODE REVIEW

File: .github/actions/image-builder/action.yml [hunk 1]
```diff
@@ -42,6 +42,10 @@ inputs:
     description: Prevent actual triggering the image builder
     required: false
     default: false
+  use-go-internal-modules:
```
Comment (Sawthis) [inline .github/actions/image-builder/action.yml:45]:
```suggestion
  use-go-sap-internal-modules:
```
maybe we should add that it is sap internal module?

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: .github/actions/image-builder/action.yml (hunk 1 above)
Comment (KacperMalachowski) [reply .github/actions/image-builder/action.yml:45]: Sounds reasonable

File: .github/actions/image-builder/action.yml [hunk 2]
```diff
@@ -42,6 +42,10 @@ inputs:
     description: Prevent actual triggering the image builder
     required: false
     default: false
+  use-go-internal-sap-modules:
+    description: Setup access to go internal SAP modules in build environment
```
Comment (Sawthis) [inline .github/actions/image-builder/action.yml:46]:
```suggestion
    description: Setup access to Go internal SAP modules in build environment
```

File: .github/actions/image-builder/action.yml [hunk 3]
```diff
@@ -81,4 +85,4 @@ runs:
     - uses: docker://europe-docker.pkg.dev/kyma-project/prod/image-builder:v20241112-fd0d9381
       id: build
       with:
-        args: --name=${{ inputs.image-name }} --context=${{ inputs.context }} --dockerfile=${{ inputs.dockerfile }} --azure-access-token=${{ inputs.ado-token }} --oidc-token=${{ inputs.oidc-token }} ${{ steps.prepare-build-args.outputs.build-args }} ${{ steps.prepare-tags.outputs.tags }} --export-tags=${{ inputs.export-tags }} --config=${{ inputs.config }} --env-file=${{ inputs.env-file }} --build-in-ado=true
\ No newline at end of file
+        args: --name=${{ inputs.image-name }} --context=${{ inputs.context }} --dockerfile=${{ inputs.dockerfile }} --azure-access-token=${{ inputs.ado-token }} --oidc-token=${{ inputs.oidc-token }} ${{ steps.prepare-build-args.outputs.build-args }} ${{ steps.prepare-tags.outputs.tags }} --export-tags=${{ inputs.export-tags }} --config=${{ inputs.config }} --env-file=${{ inputs.env-file }} --build-in-ado=true --use-go-internal-modules=${{ inputs.use-go-internal-sap-modules }}
```
Comment (Sawthis) [inline .github/actions/image-builder/action.yml:88]:
```suggestion
        args: --name=${{ inputs.image-name }} --context=${{ inputs.context }} --dockerfile=${{ inputs.dockerfile }} --azure-access-token=${{ inputs.ado-token }} --oidc-token=${{ inputs.oidc-token }} ${{ steps.prepare-build-args.outputs.build-args }} ${{ steps.prepare-tags.outputs.tags }} --export-tags=${{ inputs.export-tags }} --config=${{ inputs.config }} --env-file=${{ inputs.env-file }} --build-in-ado=true --use-go-internal-sap-modules=${{ inputs.use-go-internal-sap-modules }}
```

File: .github/workflows/image-builder.yml [hunk 4]
```diff
@@ -40,6 +40,11 @@ on:
         required: false
         type: string
         default: ""
+      use-go-internal-sap-modules:
+        description: Setup access to go internal SAP modules in build environment
```
Comment (Sawthis) [inline .github/workflows/image-builder.yml:44]:
```suggestion
        description: Setup access to Go internal SAP modules in build environment
```

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: .github/actions/image-builder/action.yml (hunk 3 above)
Comment (KacperMalachowski) [reply .github/actions/image-builder/action.yml:88]: That require changes in image itself, I will handle it in separate PR round ;)

Comment (Sawthis) [reply .github/actions/image-builder/action.yml:88]: ok

**Review by Sawthis — APPROVED** [review]

**Pull request merged**